mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala

```diff
@@ -183,11 +190,28 @@ class CrossValidator @Since("1.2.0") (@Since("1.4.0") override val uid: String)
       }
 
       // Wait for metrics to be calculated
-      val foldMetrics = foldMetricFutures.map(ThreadUtils.awaitResult(_, Duration.Inf))
-
-      // Unpersist training & validation set once all metrics have been produced
-      trainingDataset.unpersist()
-      validationDataset.unpersist()
+      val foldMetrics = try {
+        foldMetricFutures.map(ThreadUtils.awaitResult(_, Duration.Inf))
+      }
+      catch {
+        case e: Throwable =>
+          subTaskFailed = true
+          try {
+            Thread.sleep(1000)
```

Comment:
Why we need sleep 1000?

Reply:
https://github.com/apache/spark/pull/32399#discussion_r629087709